mlir/lib/CAPI/IR/BuiltinAttributes.cpp

```diff
@@ -745,6 +745,9 @@ int64_t mlirDenseElementsAttrGetInt64Value(MlirAttribute attr, intptr_t pos) {
 uint64_t mlirDenseElementsAttrGetUInt64Value(MlirAttribute attr, intptr_t pos) {
   return llvm::cast<DenseElementsAttr>(unwrap(attr)).getValues<uint64_t>()[pos];
 }
+int64_t mlirDenseElementsAttrGetIndexValue(MlirAttribute attr, intptr_t pos) {
```

Comment:
Ideally, index is target-specific, but I guess we need to iterate over DenseElementsAttr of index. Don't know how to handle it differently than what you are doing, though.

Reply:
That is true. I will open a discourse thread/PR and check what up-stream thinks. But for now we will use this to move forward.

Reply:
emitc lowers `index` into `size_t`, so I suggest to use that

Reply:
Done